relay-general/src/store/transactions/rules.rs

```diff
@@ -0,0 +1,195 @@
+use chrono::{DateTime, Utc};
+use relay_common::Glob;
+use serde::{Deserialize, Deserializer, Serialize};
+
+use crate::protocol::TransactionSource;
+
+/// Helper function to deserialize the string patter into the [`relay_common::Glob`].
+fn deserialize_glob_pattern<'de, D>(deserializer: D) -> Result<Glob, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let pattern = String::deserialize(deserializer)?;
+    let glob = Glob::builder(&pattern)
+        .capture_star(true)
+        .capture_double_star(false)
+        .capture_question_mark(false)
+        .build();
+    Ok(glob)
+}
+
+/// Default value for substitution in [`RedactionRule`].
+fn default_substitution() -> String {
+    "*".to_string()
+}
+
+/// Contains transaction attribute the rule must only be applied to.
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub struct RuleScope {
+    /// The source of the transaction.
+    pub source: TransactionSource,
+}
+
+impl Default for RuleScope {
+    fn default() -> Self {
+        Self {
+            source: TransactionSource::Url,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub struct Replace {
+    #[serde(default = "default_substitution")]
+    substitution: String,
+}
+
+/// Describes what to do with the matched pattern.
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+#[serde(tag = "method", rename_all = "snake_case")]
+pub enum RedactionRule {
+    Replace(Replace),
+    #[serde(other, skip_serializing)]
```

Comment:
```suggestion
    #[serde(other)]
```

If we skip serializing, an internal Relay passes on this config to a downstream Relay which then assumes the default substitution, which is not what we want.